server/sonar-server/src/main/java/org/sonar/server/plugins/ws/InstalledPluginsWsAction.java

```diff
@@ -0,0 +1,110 @@
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.plugins.ws;
+
+import com.google.common.base.Predicate;
+import com.google.common.io.Resources;
+import org.sonar.api.platform.PluginMetadata;
+import org.sonar.api.platform.PluginRepository;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.utils.text.JsonWriter;
+
+import java.util.Comparator;
+import java.util.List;
+
+import static com.google.common.base.Objects.firstNonNull;
+import static com.google.common.collect.Iterables.filter;
+import static com.google.common.collect.Lists.newArrayList;
+import static java.util.Collections.sort;
+
+/**
+ * Implementation of the {@code installed} action for the Plugins WebService.
+ */
+public class InstalledPluginsWsAction implements PluginsWsAction {
+  private static final String DEFAULT_VERSION = "-";
+  private static final String PROPERTY_KEY = "key";
+  private static final String PROPERTY_NAME = "name";
+  private static final String PROPERTY_VERSION = "version";
+
+  private final PluginRepository pluginRepository;
+
+  public InstalledPluginsWsAction(PluginRepository pluginRepository) {
+    this.pluginRepository = pluginRepository;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    controller.createAction("installed")
+      .setDescription("Get the list of all the plugins installed on the SonarQube instance, sorted by name")
+      .setSince("5.2")
+      .setHandler(this)
+      .setResponseExample(Resources.getResource(this.getClass(), "example-installed_plugins.json"));
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    List<PluginMetadata> pluginMetadatas = retrieveAndSortPluginMetadata();
+    writeMetadataList(response, pluginMetadatas);
+  }
+
+  private List<PluginMetadata> retrieveAndSortPluginMetadata() {
+    List<PluginMetadata> pluginMetadatas = newArrayList(filter(pluginRepository.getMetadata(), NotCorePluginsPredicate.INSTANCE));
+    sort(pluginMetadatas, PluginMetadataNameComparator.INSTANCE);
```

Comment:
using a SortedSet instead of a list will have an impact: Plugins with the same name but, for example, a different key will be considered equal and only one will appear in the JSON response.

@jblievremont, are plugin names unique ?